modules/markup/html.go

```diff
@@ -944,6 +944,13 @@ func comparePatternProcessor(ctx *RenderContext, node *html.Node) {
 			return
 		}
 
+		// Ensure that every group (m[0]...m[7]) has a match
+		for i := 0; i < 8; i++ {
```

Comment:
This is unfortunately incorrect on partial matches too.

`len(m)` needs to be checked to ensure that it is at least 8